### PR TITLE
DOCS-5555 rum feature matrix

### PR DESCRIPTION
**Pull request by rtrieu**
### What does this PR do?
Adds a compatibility feature matrix on the RUM index page.

I'm not sure if this is the best way to present this information, so open to feedback.

### Motivation
[DOCS-5555](https://datadoghq.atlassian.net/browse/DOCS-5555).

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-5555]: https://datadoghq.atlassian.net/browse/DOCS-5555?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rtrieu** [comment]
Created an Editorial Review card: https://datadoghq.atlassian.net/browse/DOCS-5637

File: content/en/real_user_monitoring/_index.md [hunk 1]
```diff
@@ -73,6 +73,46 @@ Select an application type to start collecting RUM data:
...
+| Track Views and Actions (RUM) | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} | - All actions tracked in **Flutter Web** are recorded as `custom` <br> - **Roku** only supports manual action tracking. |
+| Feature Flags tracking and release tracking | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} |  |  |
+| Error tracking and source mapping | {{< X >}} | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | - For **Flutter**, only Error Tracking is supported <br> - Only partially supported for **React Native** |
+| Crash tracking, symbolication, and deobfuscation | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} |  |
+| Stop sessions (Kiosk Monitoring) | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} |  |  |
+| Track Events in WebViews |  | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} |  |  |
+| Monitor platform-specific vitals | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} |  |  |
+| Global attribute tracking in Logs  | {{< X >}} |  |  |  |  |  |  |
+| Client side tracing |  | | {{< X >}}  |  | {{< X >}} |  |  |  |
```
Comment (fuzzybinary) [inline content/en/real_user_monitoring/_index.md:111]: Personally, I like this feature matrix way better -- just wanted to point out that the Client Side Tracing is off by 1 (says it's supported by Flutter and React Native, should be Android and iOS). Although the order of the headers should probably be changed well (Flutter should come after iOS)

Comment (rtrieu) [reply content/en/real_user_monitoring/_index.md:111]: fixed!

**Review by xgouchet — APPROVED** [review]

**Comment by fuzzybinary** [comment]
Two more minor things -- recent development, I think we can remove "- For Flutter, only Error Tracking is supported". Also I think React Native may support client side tracing @louiszawadzki for confirmation.

File: content/en/real_user_monitoring/_index.md [hunk 2]
```diff
@@ -73,6 +73,29 @@ Select an application type to start collecting RUM data:
 
 </br>
 
+### Compatibility requirements
+
+**Note**: The Datadog Flutter SDK is not supported for MacOS, Windows, or Linux.
+
+The following RUM capabilities are supported relative to each platform:
```
Comment (kayayarai) [inline content/en/real_user_monitoring/_index.md:80]:
```suggestion
The following table shows which RUM capabilities are supported on each platform:
```

File: content/en/real_user_monitoring/_index.md [hunk 3]
```diff
@@ -73,6 +73,29 @@ Select an application type to start collecting RUM data:
...
+
+| Feature                               | Browser | Android | iOS |   Flutter   | React Native | Roku | Notes |
+| ------------------------------------- | --------|---------|---------|---------|--------------|------|-------|
+| Send logs to Datadog  | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} |  |
+| Distributed tracing of network requests | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} | The **Datadog Roku SDK** is only able to track some types of HTTP requests. |
+| Track Views and Actions (RUM) | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} | - All actions tracked in **Flutter Web** are recorded as `custom` <br> - **Roku** only supports manual action tracking. |
```
Comment (kayayarai) [inline content/en/real_user_monitoring/_index.md:86]:
```suggestion
| Track Views and Actions (RUM) | {{< X >}} | {{< X >}}  | {{< X >}}  | {{< X >}} | {{< X >}} | {{< X >}} | - All actions tracked in **Flutter Web** are recorded as `custom` <br> - **Roku** supports only manual action tracking. |
```

File: content/en/real_user_monitoring/_index.md [hunk 4]
```diff
@@ -73,6 +73,29 @@ Select an application type to start collecting RUM data:
 
 </br>
 
+### Compatibility requirements
```
Comment (kayayarai) [inline content/en/real_user_monitoring/_index.md:76]:
```suggestion
### Capabilities and platform support
```

**Review by kayayarai — APPROVED** [review]
Just a few minor wording suggestions!

**Pull request merged**